terraform/modules/spack/gitlab_db.tf

```diff
@@ -30,23 +88,350 @@ module "gitlab_db" {
+  create_db_parameter_group = true
+  parameters = [
+    # Enable logical replication so CDCs can be run correctly
     {
-      from_port   = 5432
-      to_port     = 5432
-      protocol    = "tcp"
-      description = "PostgreSQL access from within VPC"
-      cidr_blocks = module.vpc.vpc_cidr_block
+      name         = "rds.logical_replication"
+      value        = "1"
+      apply_method = "pending-reboot"
     },
+    # Ensure a timeout of at least 10000 (10s)
+    {
+      name         = "wal_sender_timeout"
+      value        = "30000"
+      apply_method = "pending-reboot"
+    },
+    # Explicitly set the max number of logical replication, autovacuum, and parallel workers.
+    # The value of these parameters are stated here in order to encode them explicitly,
+    # and are the values from the existing parameter group.
+    {
+      name         = "max_logical_replication_workers",
+      value        = local.max_logical_replication_workers
+      apply_method = "pending-reboot"
+    },
+    {
+      name         = "autovacuum_max_workers",
+      value        = local.autovacuum_max_workers
+      apply_method = "pending-reboot"
+    },
+    {
+      name         = "max_parallel_workers",
+      value        = local.max_parallel_workers
+      apply_method = "pending-reboot"
+    },
+
+    # The AWS docs highlight that this value should be at least the sum of the
+    # 3 preceeding parameters, which is why they're defined here.
+    {
+      name         = "max_worker_processes"
+      value        = local.max_worker_processes
+      apply_method = "pending-reboot"
+    }
   ]
 }
+
+# The definition for the RDS instance that clones from the main gitlab RDS instance
+module "gitlab_db_clone" {
+  source  = "terraform-aws-modules/rds/aws"
+  version = "5.2.3"
+
+  identifier = "gitlab-${var.deployment_name}-clone"
+
+  engine               = "postgres"
+  engine_version       = "14.6"
+  family               = "postgres14"
+  major_engine_version = "14"
+  instance_class       = var.gitlab_db_instance_class
+
+  db_name                = "gitlabhq_production"
+  username               = "postgres"
+  port                   = "5432"
+  create_random_password = false
+  password               = local.gitlab_db_clone_master_password
+
+  publicly_accessible  = false
+  db_subnet_group_name = aws_db_subnet_group.spack.name
+
+
+  maintenance_window              = "Sun:00:00-Sun:03:00"
+  backup_window                   = "03:00-06:00"
+  enabled_cloudwatch_logs_exports = ["postgresql", "upgrade"]
+  create_cloudwatch_log_group     = true
+
+  backup_retention_period = 7
+  skip_final_snapshot     = false
+  deletion_protection     = true
+
+  allocated_storage     = 500
+  max_allocated_storage = 1000
+
+  vpc_security_group_ids = [module.postgres_security_group.security_group_id]
+}
+
+
+# Defines the Change Data Capture migration between the source and clone RDS instances
+module "database_migration_service" {
```

Comment:
Should these two modules (the new RDS one and the DMS one) be in their own file? I think it's valuable to have a single .tf file that just represents "the gitlab database", without any of the analytics stuff which is technically not "required" for gitlab to work correctly. Maybe someone who's generally unfamiliar with how the analytics clone works might want to change something about the gitlab DB, and get confused by all the JSON blocks that configure the DMS replication. Any thoughts?

Reply:
I think that's a good idea. Also, I'm realizing now that since this is at the module level it'll create it for both staging and prod, which we don't want. I can move this into a new file, and add the `count` meta argument so it only creates things for prod.

Reply:
Alternatively, you might also be able to move it into its own file in `terraform/production/xx.tf`, if this is for sure a "production-only" thing. Although if that requires adding several new outputs to the `spack` module, maybe it's not worth it, just throwing it out there. fwiw I did the `count` thing for OpenSearch because I thought maybe we may want to spin up a staging version of that someday to test upgrades, etc.

Reply:
Yeah I was trying to avoid polluting our outputs too much, but I'll see what works best.